R/data_interface.R

```diff
@@ -350,16 +350,14 @@ spark_data_write_generic <- function(df, path, fileMethod, mode = NULL, csvOptio
 #' @export
 spark_read_table <- function(sc,
                              name,
-                             path,
                              options = list(),
                              repartition = 0,
                              memory = TRUE,
                              overwrite = TRUE) {
 
   if (overwrite) spark_remove_table_if_exists(sc, name)
 
-  operation <- if (spark_version(sc) < "2.0.0") "load" else "table"
-  df <- spark_data_read_generic(sc, path, operation, options)
+  df <- spark_data_read_generic(sc, name, "table", options)
```

Comment:
@kevinushey yes, it works in Spark 1.6 but NOT in Spark 1.6 in local mode. In Spark 1.6 local you get this exception:

```
org.apache.spark.SparkException: Job aborted due to stage failure: Task 1 in stage 6.0 failed 1 times, most recent failure: Lost task 1.0 in stage 6.0 (TID 9, localhost): java.io.IOException: Mkdirs failed to create file:/user/hive/warehouse/iris_named/_temporary/0/_temporary/attempt_201703201501_0006_m_000001_0 (exists=false, cwd=file:/Users/javierluraschi/RStudio/sparklyr)
```

The problem is that somehow, Spark 1.6 does not set the right permissions on the Hive folder. I verified that `chmod g+w /user/hive/` solves this issue.

While we could fix this ourselves, I'm a bit hesitant since we currently run without workarounds in OS X (as opposed to several config changes in Windows). I'm thinking there is a set of non-core functions that are not supported in local environment, thoughts?